packages/components/src/common/dimensions.ts

```diff
@@ -16,3 +16,69 @@ export const unmeasuredDimensions: ElementDimensions = {
 export interface DimensionsById {
     [id: string]: ElementDimensions;
 }
+
+export const observeElementDimensions = (element: HTMLElement, callback: (dimensions: ElementDimensions) => void) => {
+    const observer = new ResizeObserver(() => callback(element.getBoundingClientRect()));
+
+    if (element !== null) return;
+
+    return () => observer.disconnect();
+};
+
+export const observeWindowDimensions = (callback: (dimensions: ElementDimensions) => void) => {
+    const onResize = () =>
+        callback({
+            width: window.innerWidth,
+            height: window.innerHeight,
+        });
+
+    onResize();
+
+    window.addEventListener('resize', onResize, {
+        capture: false,
+        passive: true,
+    });
+
+    return () => window.removeEventListener('resize', onResize);
+};
+
+export const getSizeFromDimensions = (dimensions: ElementDimensions, sizeAsHeight: boolean): number => {
```

Comment:
I don't like the name sizeAsHeight

Reply:
Discussed, will be executed later in stylable-components